src/mac/tap_dev.h

```diff
@@ -4,45 +4,46 @@
  * Copyright (C) 2020 Lukas Ostendorf <lukas.ostendorf@gmail.com>
  *                    and the project contributors
  *
- * This library is free software; you can redistribute it and/or modify it under the terms of the
- * GNU Lesser General Public License as published by the Free Software Foundation; version 3.0.
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; version 3.0.
  *
- * This library is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
- * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
- * See the GNU Lesser General Public License for more details.
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
  *
- * You should have received a copy of the GNU Lesser General Public License along with this library;
- * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
- * Boston, MA 02110-1301 USA
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
 #ifndef MAC_TAP_DEV_H_
 #define MAC_TAP_DEV_H_
 
-#include <sys/socket.h>
```

Comment:
There's a problem with linux/if.h. One has to include sys/socket.h **before** including linux/if.h, otherwise the included file contains incomplete struct definitions.
https://stackoverflow.com/questions/8755030/whats-wrong-with-linux-if-h-and-net-if-h

Required change:
include <sys/socket.h> at the top.

Reply:
Ok, to solve this for future clang-format runs, we have to separate this into separate include blocks, as the default style only reoders includes inside the same block. Blocks are separated by lines without "#include ..."

So for us the solution would be:
```
#include <sys/socket.h>
// must be included after socket.h
#include <linux/if.h>
#include <linux/if_tun.h>
````

Reply:
Yes, just verified that your fix passes the clang-format check and successfully compiles.